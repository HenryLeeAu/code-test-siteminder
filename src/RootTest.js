// reuse redux provider in real component and testing code
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import reducers from 'reducers';
import ReduxThunk from 'redux-thunk';

export default ({ children, initialState = {} }) => {
  const store = createStore(
    reducers,
    initialState,
    applyMiddleware(ReduxThunk)
  );

  return <Provider store={store}>{children}</Provider>;
};
