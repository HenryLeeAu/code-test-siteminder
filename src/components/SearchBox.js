import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from 'actions';
import _ from 'lodash';
class SearchBox extends Component {
  state = {
    value: '',
  };
  handleChange = e => {
    //console.log(this.props)
    this.setState(
      {
        value: e.target.value,
      },
      () => {}
    );
    this.props.inputKeyword(e.target.value);
    this.search();
  };
  search = _.debounce(() => {
    if (this.state.value.length <= 2) return;
    this.props.fetchMovieList(this.state.value);
  }, 500);
  render() {
    return (
      <div className="SearchBox">
        <input
          type="text"
          value={this.state.value}
          onChange={this.handleChange}
          placeholder="keyword"
        />
      </div>
    );
  }
}
function mapStateToProps(state) {
  return {
    searchStatus: state.searchStatus,
  };
}
export default connect(
  mapStateToProps,
  actions
)(SearchBox);
