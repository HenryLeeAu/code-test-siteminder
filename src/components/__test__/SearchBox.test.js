import React from "react";

import SearchBox from "components/SearchBox";
import { mount, unmount } from "enzyme";
import RootTest from "RootTest";

let wrappered;
let fetchMovieList =jest.fn()
beforeEach(() => {
  wrappered = mount(
    <RootTest>
      <SearchBox fetchMovieList={fetchMovieList}/>
    </RootTest>
  );
});
afterEach(() => {
  wrappered.unmount();
});

describe("Searchbox", () => {

  it("render each component", () => {
    expect(wrappered).toMatchSnapshot();
  });

  it("update text and call search function", () => {
   
    const instance = wrappered.find("SearchBox").instance();
    const search = jest.spyOn(instance, "search");
    wrappered.find("input").simulate("change", {
      target: { value: "marvel" }
    });
    wrappered.update();
    expect(search).toHaveBeenCalledTimes(1);
    expect(wrappered.find("input").prop("value")).toEqual("marvel");

  });
});
