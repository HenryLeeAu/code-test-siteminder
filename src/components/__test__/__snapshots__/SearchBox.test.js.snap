// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbox render each component 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(SearchBox)
      fetchMovieList={[MockFunction]}
    >
      <SearchBox
        decreaseCurrentPage={[Function]}
        fetchMovieDetail={[Function]}
        fetchMovieList={[Function]}
        getList={[Function]}
        increaseCurrentPage={[Function]}
        inputKeyword={[Function]}
        searchStatus={
          Object {
            "currentId": null,
            "currentMovieDetail": null,
            "currentMovieList": Array [],
            "currentPage": null,
            "keyword": "",
            "totalNum": null,
            "totalPages": null,
          }
        }
        showMovieDetail={[Function]}
        updateCurrentPageNum={[Function]}
      >
        <div
          className="SearchBox"
        >
          <input
            aria-label="Search"
            onChange={[Function]}
            placeholder="Input your keyword"
            type="text"
            value=""
          />
        </div>
      </SearchBox>
    </Connect(SearchBox)>
  </Provider>
</_default>
`;
